docs/learn/dapp-developer/read-asset-data.md

```diff
@@ -180,9 +180,14 @@ Based on this information, the asset metadata is fetched in different ways.
 
 ## Fetch the Asset Metadata
 
-To fetch the whole contracts JSON file, you can use the [`fetchData('LSP4Metadata')`](../../tools/erc725js/classes/ERC725.md#fetchdata) directly on the asset address. Please check on the [LSP4 Digital Asset Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata/#types-of-digital-assets) page where to fetch the metadata. In the case of a collection, you will have to fetch the `LSP4Metadata` for each token address within the collection.
+As described on the different asset types of the [LSP4 Digital Asset Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata.md#types-of-digital-assets) documentation, metadata can be attached:
 
-<!-- We might need to make it more clear for each token type how to fetch the data and get the tokenId etc. -->
+- As **global token information** of the contract (Token or LSP7 NFT)
+- To each **individual token ID** (LSP8 NFT or Collection)
```

Comment:
lets also make sure we include baseUri https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenmetadatabaseuri